server/schedulers/grant_hot_region.go

```diff
@@ -0,0 +1,376 @@
+	return schedule.EncodeConfig(s.conf)
+}
+
+func (s *grantHotRegionScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
+	regionAllowed := s.OpController.OperatorCount(operator.OpRegion) < cluster.GetOpts().GetRegionScheduleLimit()
+	leaderAllowed := s.OpController.OperatorCount(operator.OpLeader) < cluster.GetOpts().GetLeaderScheduleLimit()
+	if !regionAllowed {
+		operator.OperatorLimitCounter.WithLabelValues(s.GetType(), operator.OpRegion.String()).Inc()
+	}
+	if !leaderAllowed {
+		operator.OperatorLimitCounter.WithLabelValues(s.GetType(), operator.OpLeader.String()).Inc()
+	}
+	return regionAllowed && leaderAllowed
+}
+
+func (s *grantHotRegionScheduler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	s.handler.ServeHTTP(w, r)
+}
+
+type grantHotRegionHandler struct {
+	rd     *render.Render
+	config *grantHotRegionSchedulerConfig
+}
+
+func (handler *grantHotRegionHandler) UpdateConfig(w http.ResponseWriter, r *http.Request) {
+	var input map[string]interface{}
+	if err := apiutil.ReadJSONRespondError(handler.rd, w, r.Body, &input); err != nil {
+		return
+	}
+	ids, ok := input["store-id"].(string)
+	if !ok {
+		_ = handler.rd.JSON(w, http.StatusBadRequest, errs.ErrSchedulerConfig)
+		return
+	}
+	storeIDs := make([]uint64, 0)
+	for _, v := range strings.Split(ids, ",") {
+		id, err := strconv.ParseUint(v, 10, 64)
+		if err != nil {
+			_ = handler.rd.JSON(w, http.StatusBadRequest, errs.ErrBytesToUint64)
+			return
+		}
+		storeIDs = append(storeIDs, id)
+	}
+	leaderID, err := strconv.ParseUint(input["store-leader-id"].(string), 10, 64)
+	if err != nil {
+		_ = handler.rd.JSON(w, http.StatusBadRequest, errs.ErrBytesToUint64)
+		return
+	}
+	if !handler.config.setStore(leaderID, storeIDs) {
+		_ = handler.rd.JSON(w, http.StatusBadRequest, errs.ErrSchedulerConfig)
+		return
+	}
+
+	if err = handler.config.Persist(); err != nil {
+		handler.config.StoreLeadID = 0
+		_ = handler.rd.JSON(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	_ = handler.rd.JSON(w, http.StatusOK, nil)
+}
+
+func (handler *grantHotRegionHandler) ListConfig(w http.ResponseWriter, _ *http.Request) {
+	conf := handler.config.Clone()
+	_ = handler.rd.JSON(w, http.StatusOK, conf)
+}
+
+func newGrantHotRegionHandler(config *grantHotRegionSchedulerConfig) http.Handler {
+	h := &grantHotRegionHandler{
+		config: config,
+		rd:     render.New(render.Options{IndentJSON: true}),
+	}
+	router := mux.NewRouter()
+	router.HandleFunc("/config", h.UpdateConfig).Methods(http.MethodPost)
+	router.HandleFunc("/list", h.ListConfig).Methods(http.MethodGet)
+	return router
+}
+
+func (s *grantHotRegionScheduler) Schedule(cluster opt.Cluster) []*operator.Operator {
+	schedulerCounter.WithLabelValues(s.GetName(), "schedule").Inc()
+	i := s.r.Int() % len(s.types)
+	return s.dispatch(s.types[i], cluster)
+}
+
+func (s *grantHotRegionScheduler) dispatch(typ rwType, cluster opt.Cluster) []*operator.Operator {
```

Comment:
Will it conflict with other schedulers?

Reply:
yes,it will conflict with balance hot region scheduler